### DISCUSSION TITLE
请问类型是 group 的私人消息是怎么产生的？

### DISCUSSION THREAD

**Discussion by Arondight** [takayama-lily/oicq#277] [Q&A] [answered]
@takayama-lily 

```
recv from: [Private: 2057706667(group)]
Failed to send: [Private: 2057706667] 发送失败，请先添加对方为好友(16)
```

**Comment by Arondight** [comment, marked as answer]
我找了半天，发现是一个陌生人，现在 `sendPrivateMsg` 不能给陌生人发消息了，但是事件还是 `message.private`

**Reply by Arondight** [reply]
我看了下代码，似乎除了快速回复以外没有给陌生人发消息的直接方法？

**Reply by Arondight** [reply]
我打印了一下陌生人和好友聊天，唯一的区别就是 `message.sender.group_id` 的有无

**Reply by takayama-lily** [reply]
`sendPrivateMsg`不支持发送群临时会话了，恩，这个似乎changelog里漏写了。
发群临时会话可以使用`client.sendTempMsg()`或`member.sendMsg()`或`msg.reply()`。
（之前的内部实现是：遍历所有的群，找到该成员然后发送，觉得不合适就去掉了）

**Reply by Arondight** [reply]
额……看样子我需要 bind 一下重写一个 `sendPrivateMsg` ？有考虑加个默认关闭的参数吗来支持这个特性吗

**Reply by Arondight** [reply]
哦，我在外面也能处理这个，简单的查一下好友列表，没有的话查群调用 `sendTempMsg` 就可以了

**Reply by takayama-lily** [reply]
> 额……看样子我需要 bind 一下重写一个 `sendPrivateMsg` ？有考虑加个默认关闭的参数吗来支持这个特性吗

不会再支持了，主要是不传群号仅靠遍历群的话会有问题，比如一个人同时在多个群，其中有的群可能关闭了临时会话。
（新版本是为了向前看，并彻底抛弃旧包袱，所以难免会有一些牺牲）

**Reply by Arondight** [reply]
好的，感谢，我自己处理一下就行
